src/app/Dashboard/AddTopic.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import './AddTopic.css';
+export default class AddTopic extends React.Component {
+  state = {
+    isHidden: true
+  };
+
+  toggleHidden = () => {
+    this.setState({
+      isHidden: !this.state.isHidden
+    });
+  };
+
+  onSubmit(e) {
+    e.preventDefault();
+    console.log('create new topic with name:', e.target.folderName.value);
+    this.toggleHidden();
+  }
+
+  render() {
+    return (
+      <>
+        <button onClick={() => this.toggleHidden()}>+ add topic</button>
```

Comment:
By inlining your onClick function this way, it creates a new anonymous function every time a new instance of this component is created. 

Instead, we should reference it like `onClick={this.toggleHidden}`. This coincides with how Janet and I have been writing our class components. Instead of creating a new callback function in place every time, it will now simply call the callback `this.toggleHidden` every time the event fires.

This will work because the function `toggleHidden` is now an arrow function, so it doesn't create its own `this` context like a normal function.